pkg/miner/internal.go

```diff
@@ -10,6 +10,12 @@ func MineBlock(version proto.BlockVersion, nxt proto.NxtConsensus, pair proto.Ke
 	if err != nil {
 		return nil, err
 	}
+	if b.Version >= proto.ProtoBlockVersion {
+		err := b.SetTransactionsRoot(scheme)
```

Comment:
Can we move `SetTransactionRoot()` call to `CreateBlock()` function?
It is used in multiple places, so might be better this way.